python/ray/tests/test_basic_5.py

```diff
@@ -0,0 +1,82 @@
+# coding: utf-8
+import gc
+import logging
+import os
+import sys
+import time
+
+import pytest
+
+import ray.cluster_utils
+from ray._private.test_utils import wait_for_pid_to_exit
+
+import ray
+
+logger = logging.getLogger(__name__)
+
+
+def test_background_tasks_with_max_calls(shutdown_only):
+    ray.init(
+        # TODO (Alex): We need to fix
+        # https://github.com/ray-project/ray/issues/20203 to remove this flag.
+        num_cpus=2,
+        _system_config={"worker_cap_initial_backoff_delay_ms": 0})
+
+    num_tasks = 3 if sys.platform == "win32" else 10
```

Comment:
@pcmoritz @wuisawesome I have kept `num_tasks = 3` for windows only.

Reply:
Thanks, sounds good!

Reply:
So, here are some points regarding behaviour of `test_basic_5::test_background_tasks_with_max_calls`. I changed the values of `num_cpus` in `ray.init`, `num_tasks` and modified the test a little bit to record times of each part (see bottom.)

1. `num_cpus=2000` and `num_tasks=10`

```bash
DEBUG part 1:  20.71898913383484
DEBUG part 2:  0.0
DEBUG ray.get 1:  24.325945615768433
DEBUG ray.get 2:  1.1143145561218262
DEBUG part 3:  0.0
2021-12-24 12:06:25,060 WARNING worker.py:1257 -- WARNING: 16 PYTHON worker processes have been started on node: 741f3ec400824b8d67e236e669a608aab8468d328c4b9f74405cddb1 with address: 127.0.0.1. This could be a result of using a large number of actors, or due to tasks blocked in ray.get() calls (see https://github.com/ray-project/ray/issues/3644 for some discussion of workarounds).
DEBUG ray.get 3:  27.93891668319702
DEBUG ray.get 4:  8.863697290420532
DEBUG Total time:  83.23906683921814
```

2. `num_cpus=2` and `num_tasks=10`
```bash
DEBUG part 1:  18.739073991775513
DEBUG part 2:  0.0
(scheduler +28s) Tip: use `ray status` to view detailed cluster status. To disable these messages, set RAY_SCHEDULER_EVENTS=0.
(scheduler +28s) Warning: The following resource request cannot be scheduled right now: {'CPU': 1.0}. This is likely due to all cluster resources being claimed by actors. Consider creating fewer actors or adding more nodes to this Ray cluster.
2021-12-24 12:24:17,639 WARNING worker.py:1257 -- The actor or task with ID c5f3b41ef51bed2569501b253816a14464a3da0f01000000 cannot be scheduled right now. You can ignore this message if this Ray cluster is expected to auto-scale or if you specified a runtime_env for this actor or task, which may take time to install.  Otherwise, this is likely due to all cluster resources being claimed by actors. To resolve the issue, consider creating fewer actors or increasing the resources available to this Ray cluster.
Required resources for this actor or task: {CPU: 1.000000}
Available resources on this node: {0.000000/2.000000 CPU, 1.477721 GiB/1.477721 GiB memory, 0.738860 GiB/0.738860 GiB object_store_memory, 1.000000/1.000000 node:127.0.0.1}
 In total there are 1 pending tasks and 0 pending actors on this node.
DEBUG ray.get 1:  26.28814697265625
DEBUG ray.get 2:  5.17179536819458
DEBUG part 3:  0.0
(scheduler +1m3s) Warning: The following resource request cannot be scheduled right now: {'CPU': 1.0}. This is likely due to all cluster resources being claimed by actors. Consider creating fewer actors or adding more nodes to this Ray cluster.
2021-12-24 12:24:44,700 WARNING worker.py:1257 -- WARNING: 8 PYTHON worker processes have been started on node: 6ece21740861cd21634998eebdb3eb99515655faecbbd0e95a0fd555 with address: 127.0.0.1. This could be a result of using a large number of actors, or due to tasks blocked in ray.get() calls (see https://github.com/ray-project/ray/issues/3644 for some discussion of workarounds).
2021-12-24 12:24:54,104 WARNING worker.py:1257 -- WARNING: 10 PYTHON worker processes have been started on node: 6ece21740861cd21634998eebdb3eb99515655faecbbd0e95a0fd555 with address: 127.0.0.1. This could be a result of using a large number of actors, or due to tasks blocked in ray.get() calls (see https://github.com/ray-project/ray/issues/3644 for some discussion of workarounds).
2021-12-24 12:24:59,252 WARNING worker.py:1257 -- WARNING: 12 PYTHON worker processes have been started on node: 6ece21740861cd21634998eebdb3eb99515655faecbbd0e95a0fd555 with address: 127.0.0.1. This could be a result of using a large number of actors, or due to tasks blocked in ray.get() calls (see https://github.com/ray-project/ray/issues/3644 for some discussion of workarounds).
DEBUG ray.get 3:  30.403132200241092021-12-24 12:25:06,123      WARNING worker.py:1257 -- WARNING: 14 PYTHON worker processes have been started on node: 6ece21740861cd21634998eebdb3eb99515655faecbbd0e95a0fd555 with address: 127.0.0.1. This could be a result of using a large number of actors, or due to tasks blocked in ray.get() calls (see https://github.com/ray-project/ray/issues/3644 for some discussion of workarounds).

DEBUG ray.get 4:  10.107643127441406
DEBUG Total time:  90.95563578605652
```
3. `num_cpus=2` and `num_tasks=3`

```bash
DEBUG part 1:  17.204681873321533
DEBUG part 2:  0.0
(scheduler +25s) Tip: use `ray status` to view detailed cluster status. To disable these messages, set RAY_SCHEDULER_EVENTS=0.
(scheduler +25s) Warning: The following resource request cannot be scheduled right now: {'CPU': 1.0}. This is likely due to all cluster resources being claimed by actors. Consider creating fewer actors or adding more nodes to this Ray cluster.
DEBUG ray.get 1:  6.417951345443726
DEBUG ray.get 2:  4.502125263214111
DEBUG part 3:  0.0
DEBUG ray.get 3:  9.142998456954956
DEBUG ray.get 4:  5.96869158744812
DEBUG Total time:  43.296196699142456
```

4. `num_cpus=2000` and `num_tasks=3`

```bash
DEBUG part 1:  21.341447114944458
DEBUG part 2:  0.0
DEBUG ray.get 1:  1.3339462280273438
DEBUG ray.get 2:  7.307475328445435
DEBUG part 3:  0.0
DEBUG ray.get 3:  0.8497307300567627
DEBUG ray.get 4:  8.178425550460815
DEBUG Total time:  39.20774245262146
``` 

Things I noticed,

1. The startup time of processes in total is < 1s in both the cases. Specifically 26 processes were started and it took < 1s in total for all of these processes to start. So startup time is not a problem IMO.
2. In the second case i.e., `num_cpus = 2` and `num_tasks=10`, timings vary a lot ( > 90 and <= 120s) but for `num_cpus = 2000` (or for very high values) and `num_tasks=10`, timings settle around 85s or so. 
3. For `num_tasks=3`, you can see that with `num_cpus=2000`, `ray.get 3` took barely any time (0.8s). However with `num_cpus=2`, `ray.get 3` took 6s.

Reply:
Please let me know if I should open a new issue for this.